From 0cb9c142b21ff86b6d1bb9fc1151d54fc20d1412 Mon Sep 17 00:00:00 2001
From: Priyesh <ppadmavilasom@users.noreply.github.com>
Date: Thu, 30 Mar 2017 11:35:06 -0700
Subject: [PATCH] Update security.c

Noticed when this issue was triggered in gcc-6.3
../lwmsg/src/security.c:108:28: error: self-comparison always evaluates to false [-Werror=tautological-compare]
 else if (token->tclass != token->tclass)
---
 lwmsg/src/security.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lwmsg/src/security.c b/lwmsg/src/security.c
index 6fc8cd3..a78ff72 100644
--- a/lwmsg/src/security.c
+++ b/lwmsg/src/security.c
@@ -105,7 +105,7 @@ lwmsg_security_token_equal(
     {
         return LWMSG_TRUE;
     }
-    else if (token->tclass != token->tclass)
+    else if (token->tclass != other->tclass)
     {
         return LWMSG_FALSE;
     }
